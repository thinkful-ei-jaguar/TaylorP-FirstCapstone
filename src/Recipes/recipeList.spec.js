import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import RecipeList from './recipeList'
import RecipeListContext from '../Context/recipeListContext'

describe('<RecipeList />', () => {
  it('renders without crashing', () => {
    const match = {params: {id: 1}}
    const testRecipes = [
      {
        id: 1,
        recipe_img: 'no img',
        recipe_ingredients: '2oz test,1oz test,.75 test',
        recipe_name: 'Test Recipe 1',
        recipe_prep: 'shake the stuff, do the tests',
        spirit_cat: 'testSpirit'
      },
      {
        id: 2,
        recipe_img: 'no img',
        recipe_ingredients: '2oz test,1oz test,.75 test',
        recipe_name: 'Test Recipe 2',
        recipe_prep: 'shake the stuff, do the tests',
        spirit_cat: 'OtherTestSpirit'
      },
      {
        id: 3,
        recipe_img: 'no img',
        recipe_ingredients: '2oz test,1oz test,.75 test',
        recipe_name: 'Test Recipe 3',
        recipe_prep: 'shake the stuff, do the tests',
        spirit_cat: 'AnotherTestSpirit'
      }
    ]
    const ctxval = {
      recipeList: testRecipes,
      error: null,
      setError: () => {},
      clearError: ()=> {},
      setRecipeList: () => {}
    }
    const tree = renderer.create(
      <MemoryRouter>
        <RecipeListContext.Provider value={ctxval}>
          <RecipeList match={match} />
        </RecipeListContext.Provider>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})