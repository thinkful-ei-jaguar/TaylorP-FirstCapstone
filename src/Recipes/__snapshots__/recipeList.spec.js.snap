// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> renders without crashing 1`] = `
<ul
  aria-live="polite"
  className="recipe-list"
>
  <a
    href="/recipes/1"
    onClick={[Function]}
  >
    <li
      className="recipe-li"
    >
      <h2>
        Test Recipe 1
      </h2>
    </li>
  </a>
  <a
    href="/recipes/2"
    onClick={[Function]}
  >
    <li
      className="recipe-li"
    >
      <h2>
        Test Recipe 2
      </h2>
    </li>
  </a>
  <a
    href="/recipes/3"
    onClick={[Function]}
  >
    <li
      className="recipe-li"
    >
      <h2>
        Test Recipe 3
      </h2>
    </li>
  </a>
</ul>
`;
